tests/unit/cli/test_cli_config_file.py

```diff
@@ -236,3 +243,58 @@ def test_get_ctx_defaults_nested(self):
         get_ctx_defaults("intent-answer", provider, mock_context4, "default")
 
         provider.assert_called_with(None, "default", ["local", "generate-event", "alexa-skills-kit", "intent-answer"])
+
+    def test_save_command_line_args(self):
```

Comment:
Can we split this test up into potentially 4 tests:
- Test missing `save-param` option
- Test that things get saved correctly
- Test non-CLI options aren't saved
- Test that the excluded parameters (config and save params) aren't saved.

That way should anything change, it'll be easier to update the correct test(s)

Reply:
We could, though I feel like keeping everything in one test means we can check for multiple things without duplicating as much code, since things like setup and checking will look similar for each run.
Though I would be in favour of splitting off the case of `save_params = False` as its own test, since that's different behaviour entirely.